### PR TITLE
[Free Trial - Local notification] Replace local feature flag with remote

### PR DESCRIPTION
**Pull request by selanthiraiyan**
Part of: #10092

## Description

Replaces the temporary local feature flag with the new `woo_notification_24h_after_free_trial_subscribed` remote feature flag. 

#### Prerequisite
- Since the remote feature flag Diff (D114800-code) is still not deployed, some code changes are required to test this PR:
- We can return `true` from this line. https://github.com/woocommerce/woocommerce-ios/blob/trunk/WooCommerce/Classes/Notifications/LocalNotificationScheduler.swift#L51

#### Steps
- For easier testing, please run the app on a physical device.
- Launch the app, log in, and create a new free trial site from the site picker.
- After successfully landing on the "My Store" page, kill the app.
- Open Settings, and change the time of the device to 23 hr 58 mins from the current time ( which is the free trial subscription time). 
- Wait for 2 minutes, and then you should see a notification reminding to purchase a plan. Tap on the notification, the app should be open, and a web view should be presented to upgrade your plan.

## Screenshots

24 hrs notification from the notification center

Upon tapping the notification

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: Networking/Networking/Remote/FeatureFlagRemote.swift [hunk 1]
```diff
@@ -31,6 +31,7 @@ public enum RemoteFeatureFlag: Decodable {
     case oneDayAfterStoreCreationNameWithoutFreeTrial
     case oneDayBeforeFreeTrialExpiresNotification
     case oneDayAfterFreeTrialExpiresNotification
+    case twentyFourHoursAfterFreeTrialSubscribed
```
Comment (jaclync) [inline Networking/Networking/Remote/FeatureFlagRemote.swift:34]: nit: how about `oneDayAfterFreeTrialSubscribed` to be consistent with the cases above?

**Review by jaclync — COMMENTED** [review]
The code LGTM, but I haven't got a chance to test it on a physical device. I saw that you're AFK Thursday, so I'll complete the testing and finish the review on Thursday.

**Review by jaclync — APPROVED** [review]
Testing ✅ :shipit: 

Suggestion unrelated to this PR:

After successfully upgrading to a paid plan in the upgrade flow, the free trial banner is still shown on the My store tab. Not sure if it's worth refreshing the free trial banner after dismissing the upgrade view (even better - when we know the merchant has upgraded).

File: Networking/Networking/Remote/FeatureFlagRemote.swift (hunk 1 above)
Comment (selanthiraiyan) [reply Networking/Networking/Remote/FeatureFlagRemote.swift:34]: There is a slight difference between these cases. 

- The expiry date always has midnight time as the time component. 
- The subscribed date has a specific value for the time component, meaning that the `twentyFourHoursAfterFreeTrialSubscribed` notification will be sent exactly after 24 hours after the free trial subscription. 

To show this difference I preferred to keep "24 four hours" in the naming.

**Comment by selanthiraiyan** [comment]
> After successfully upgrading to a paid plan in the upgrade flow, the free trial banner is still shown on the My store tab. Not sure if it's worth refreshing the free trial banner after dismissing the upgrade view (even better - when we know the merchant has upgraded).

Thanks for reporting this. I will work on this in a different PR. Logged an issue here https://github.com/woocommerce/woocommerce-ios/issues/10127

**Pull request merged**